src/pyradiosky/__init__.py

```diff
@@ -0,0 +1,42 @@
+"""Define namespace."""
+from importlib.metadata import PackageNotFoundError, version
+from pathlib import Path
+
+from setuptools_scm import get_version
+
+
+# copy this function here from setup.py.
+# Copying code is terrible, but it's better than altering the python path in setup.py.
+def branch_scheme(version):  # pragma: nocover
+    """
+    Local version scheme that adds the branch name for absolute reproducibility.
+
+    If and when this is added to setuptools_scm this function can be removed.
+    """
+    if version.exact or version.node is None:
+        return version.format_choice("", "+d{time:{time_format}}", time_format="%Y%m%d")
+    else:
+        if version.branch == "main":
+            return version.format_choice("+{node}", "+{node}.dirty")
+        else:
+            return version.format_choice("+{node}.{branch}", "+{node}.{branch}.dirty")
```

Comment:
Yeah, so if this branch scheme existed in a different package, we could just use it from there. Not sure if that would work OK with a `pyproject.toml`-only setup (even if they do add custom branch support to the pyproject format)..

Reply:
My understanding is that when the next version of setuptools_scm comes out, it can be in a separate file in the package and it will work properly, so this duplicating of code is hopefully temporary. We could put it in another package but then it's another dependency to manage.